sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -2019,16 +2020,79 @@ def write_pipeline_spec_to_file(
     pipeline_description: Union[str, None],
     platform_spec: pipeline_spec_pb2.PlatformSpec,
     package_path: str,
+    kubernetes_manifest_format: bool = False,
+    kubernetes_manifest_options: Optional[KubernetesManifestOptions] = None,
 ) -> None:
     """Writes PipelineSpec into a YAML or JSON (deprecated) file.
 
     Args:
         pipeline_spec: The PipelineSpec.
         pipeline_description: Description from pipeline docstring.
         package_path: The path to which to write the PipelineSpec.
-        platform_spec: The PlatformSpec.
+        kubernetes_manifest_format: Whether to output as Kubernetes manifest.
+        kubernetes_manifest_options: KubernetesManifestOptions object with manifest options. Only used if kubernetes_manifest_format is True.
     """
+    opts = kubernetes_manifest_options or KubernetesManifestOptions()
+    pipeline_name = opts.pipeline_name
+    pipeline_display_name = opts.pipeline_display_name
+    pipeline_version_name = opts.pipeline_version_name
+    pipeline_version_display_name = opts.pipeline_version_display_name
+    namespace = opts.namespace
+    include_pipeline_manifest = opts.include_pipeline_manifest
+
     pipeline_spec_dict = json_format.MessageToDict(pipeline_spec)
+
+    if kubernetes_manifest_format:
+        pipeline_name = pipeline_name or pipeline_spec.pipeline_info.name
+        pipeline_display_name = pipeline_display_name or pipeline_name
+        pipeline_version_name = pipeline_version_name or pipeline_name
+        pipeline_version_display_name = pipeline_version_display_name or pipeline_display_name
```

Comment:
Let's prioritize pipeline version name as the default for pipeline_version_display_name if it was provided.
```suggestion
        pipeline_version_display_name = pipeline_version_display_name or pipeline_version_name or pipeline_display_name
        pipeline_version_name = pipeline_version_name or pipeline_name
```